crates/flox/src/commands/channel.rs

```diff
@@ -132,62 +160,232 @@ fn construct_search_params(search_term: &str, flox: &Flox) -> Result<SearchParam
         };
         inputs.insert(entry.from.id.to_string(), input);
     }
-    let registry = Registry {
-        inputs,
-        priority,
-        defaults: RegistryDefaults::default(),
-    };
-
-    // We've already checked that the search term is Some(_)
-    let query = Query::from_str(search_term)?;
+    (inputs, priority)
+}
 
-    Ok(SearchParams {
-        registry,
-        query,
-        ..SearchParams::default()
-    })
+/// An intermediate representation of a search result used for rendering
+#[derive(Debug, PartialEq, Clone)]
+struct DisplayItem {
+    /// The input that the package came from
+    input: String,
+    /// The displayable part of the package's attribute path
+    package: String,
+    /// The package description
+    description: Option<String>,
+    /// Whether to join the `input` and `package` fields when rendering
+    join: bool,
 }
 
-// This is likely to change significantly after the output format of search results is specced out
-fn render_search_results(search_results: &SearchResults) -> Result<()> {
-    let summarized_results = search_results
+fn render_search_results_user_facing(mut search_results: SearchResults) -> Result<()> {
+    // Nothing to display
+    if search_results.results.is_empty() {
+        return Ok(());
+    }
+    // Search results contain a lot of information, but all we need for rendering are
+    // the input, the package subpath (e.g. "python310Packages.flask"), and the description.
+    let display_items = search_results
         .results
-        .iter()
+        .drain(..)
```

Comment:
That's just not yet rebased on the other or right?

Reply:
Right, I'm not adding any new `drain`s. Part of the problem with getting ahead of review.